cache/v0.2/cache-v0.2.graphql

```diff
@@ -0,0 +1,85 @@
+"""
+Possible values for the `@cacheControl` `scope` argument (unused on the client).
+"""
+enum CacheControlScope {
+    PUBLIC
+    PRIVATE
+}
+
+"""
+Configures cache settings for a field or type.
+
+- `maxAge`: The maximum amount of time the field's cached value is valid, in seconds.
+- `inheritMaxAge`: If true, the field inherits the `maxAge` of its parent field. If set to `true`, `maxAge` must not be provided.
+- `scope`: Unused on the client.
+
+When applied to a type, the settings apply to all schema fields that return this type.
+Field-level settings override type-level settings.
+
+For example:
+
+```graphql
+type Query {
+    me: User
+    user(id: ID!): User @cacheControl(maxAge: 5)
+}
+
+type User @cacheControl(maxAge: 10) {
+    id: ID!
+    email: String
+}
+```
+`Query.me` is valid for 10 seconds, and `Query.user` for 5 seconds.
+"""
+directive @cacheControl(
+    maxAge: Int
+    inheritMaxAge: Boolean
+    scope: CacheControlScope
+) on FIELD_DEFINITION | OBJECT | INTERFACE | UNION | QUERY | MUTATION | SUBSCRIPTION | FRAGMENT_DEFINITION
```

Comment:
Is the `inheritMaxAge` parameter not compatible with applying the directive to `OBJECT/INTERFACE` types?

Reply:
Let's use an example:

```graphql
type Query {
  authors: [Author!]!
}

type Author @cacheControl(maxAge: 3600) {
  books: [Book!]!
}

type Book @cacheControl(inheritMaxAge: true) {
  id: ID!
}
```
```graphql
query GetAuthorsWithBooks {
  authors { # 3600 - Because it's of type Author which has maxAge 3600
    id # 3600 - Because it's a scalar so inherits its parent field's maxAge
    books { # 3600 - Since it's not a scalar, it would be the default value (0),
            # but there's a maxAge specified on Book, which is to inherit from its parent
            # field's maxAge, which in this case is 3600
      id # 3600 - Because it's a scalar so inherits its parent field's maxAge
    }
  }
}
```